include/bout/petsclib.hxx

```diff
@@ -109,11 +119,13 @@ private:
 
 class PetscLib {
 public:
-  PetscLib() {}
+  PetscLib(Options* UNUSED(opt) = nullptr) {}
   ~PetscLib() {}
   
   static void setArgs(int &UNUSED(c), char** &UNUSED(v)) {}
   
+  KSP createKSPWithOptions(MPI_Comm& comm);
+
   static void cleanup() {}
 };
```

Comment:
I thought, but might be wrong, that we tried to ensure the PetscLib was a singleton type object. I'm probably wrong but if I'm not then I think that might cause issues with having a top level `prefix_string`. 

As I wrote this I became less convinced that  it is a singleton so I'm probably just wrong.

Reply:
It's referenced counted rather than a singleton. The last remaining instance will cleanup PETSc.

Reply:
Before I started there was already a `static int count; ///< How many instances?`, so not a singleton (although I found it surprising that it wasn't! Did turn out to be convenient though...).